lib/response_handlers/foodHandlers.js

```diff
@@ -1,10 +1,59 @@
 const $ = require('jquery')
+const Ajax = require('../ajax_requests/ajax')
 
-module.exports = class HTML {
+module.exports = class FoodHandlers {
   static populateFoodTable(data){
     data.forEach(object => {
       let row = "<tr class=food item-" + object.id + "><td data-id=" + object.id +" name=" + object.name.replace(/\s/g, '-') +" data-type='name' class='display' contenteditable='true'>" + object.name + "</td><td data-id=" + object.id +" name=" + object.calories + " data-type='calories' class='display' contenteditable='true'><span class='killer' contenteditable='false'><i class='fa fa-minus-circle' aria-hidden='true' style='font-size:24px'></i></span>" + object.calories + "</td></tr>"
       $(".food-body").prepend(row)
     })
   }
+
+  static validateFoodInput(name, calories){
+    if (name.length === 0) {
+      event.preventDefault();
+      $("p#submit-food-name").append("Please enter a food name.")
+    } else if (calories.length === 0) {
+      event.preventDefault();
+      $("p#submit-calories").append("Please enter a calorie amount.")
+    } else {
+      $("p#submit-food-name").remove()
+      $("p#submit-calories").remove()
+      Ajax.postFoods(name, calories)
+      $('input').val('').removeAttr('checked').removeAttr('selected');
+    }
+  }
+
+  static editFood(input, newFoodValue, currentFoodValue, objectId){
+    if ((newFoodValue != currentFoodValue) && (event.target.attributes[2].value === "name")){
```

Comment:
Don't think you will have access to `event` in here. 

One way of doing it could be
```
let calories, name; 

if ((newFoodValue != currentFoodValue) && (event.target.attributes[2].value === "name")){
  name = input
  calories = event.target.nextElementSibling.textContent
} else {
  name = event.target.previousElementSibling.textContent
  calories = input
}

const food = { food: { name, calories}}
Ajax.updateFood(food, objectId)
```

I might change the event and listen on the field you are updating i.e. name or calories and have 2 separate functions for them `editFoodName` `editFoodCalories`

Reply:
👍nice!!!